### PR TITLE
semaphore: Run Semaphore CI unit tests with ASAN enabled

### PR DESCRIPTION
**Pull request by stefwalter**
This helps us find bugs that valgrind misses.

https://github.com/google/sanitizers/wiki/AddressSanitizer

### CODE REVIEW

**Comment by martinpitt** [comment]
Some notekeeping:
 * Semaphore runs Ubuntu 14.04 (trusty) which has a much older toolchain. Building cockpit in a local trusty chroot and running `make check-memory` reproduces the "Shadow memory range interleaves with an existing memory mapping" errors.

 * When I build with `--enable-debug --enable-asan` in F25, I get tons of errors due to memleaks in glib during `make check`. Running with `G_SLICE=always-malloc G_DEBUG=gc-friendly` makes no difference. These don't appear in semaphore/trusty building.

 * These memleaks appear in `make check-memory` in F25 too, but later on the test just hangs.

 * I did the same build as in the first point in Ubuntu zesty (current development series), and the memleaks and eternal hang are exactly the same as in F25. This seems to confirm that this is
 a function of older/newer compiler, and not really distro specific.

So in conclusion, this doesn't currently report something useful and requires a lot more work, so I propose we put this on the back burner for now.

**Comment by stefwalter** [comment]
I agree. Lets close this.
